rb/spec/integration/selenium/webdriver/network_spec.rb

```diff
@@ -56,7 +58,9 @@ module WebDriver
       it 'adds a request handler' do
         reset_driver!(web_socket_url: true) do |driver|
           network = described_class.new(driver)
-          network.add_request_handler
+          network.add_request_handler { |event| network.continue_with_request(request_id: network.fetch_id(event)) }
```

Comment:
I roughly recall our conversation about high-level BiDi API, but I think the idea was about something similar to the following:

```ruby
# filter is essentially a list of URL patterns that should be intercepted
network.add_request_handler(filter) do |request|
  # ... here you can mutate the request as you wish
  request.headers['My-Custom-Header'] = 'Value'

  request.continue # this would continue the original (or mutated) request
  # or
  request.fail     # this would fail the request
end
```

@diemol @pujagani @titusfortner Am I recalling this correctly?

Reply:
I saw the filter implementation on the Java bindings, so I can try to translate that to Ruby now and see if that aligns with the idea from the conversations

Is there any docs or PR that I can read? maybe I skipped the one talking about filters :)

Reply:
I don't think there is any implementation yet, but a filter in Java would probably be a good start. Maybe too much for Ruby and we can just use regexp instead, I suggest you keep it simple and straightforward.

Reply:
@p0deje Your memory serves right. I have implemented what I understood from our conversations in Java/ @aguspe Appreciate your help here, but feel free to pick an approach that is most suited for the language.